### ISSUE TITLE
AmazonPay render issue with Amazon Pay v2 and checkout-sdk v1.298.2

### ISSUE THREAD

**Issue by spire-mike** [bigcommerce/checkout-sdk-js#1652]
I recently upgraded checkout-sdk from v1.115.0 to v1.298.2. My AmazonPay credentials were also recently upgraded from v1 to v2. I am noticing that on this new version, the AmazonPay button uses the `container` element rather than adding its content fully **within** the container element. The result of this that I can no longer initialize a button again after it's been deinitialized, because deinitializing removes the `container` itself from the DOM.

For example, I have a div like this in my DOM:

```
<div id="amazonRenderContainer" class="renderContainer"></div>
```
In my code, I initialize an AmazonPay button like this:
```
this.service.initializeCustomer({
  methodId: 'amazonpay',
  amazonpay: {
    container: 'amazonRenderContainer',
  },
});
```
Later, I deinitialize the button:
```
this.service.deinitializeCustomer({ methodId: 'amazonpay'});
```
My `amazonRenderContainer` gets removed from the DOM, and when I try to initialize again later in the code, I get an error:
```
InvalidArgumentError: Unable to render the Amazon Pay button to an invalid HTML container element.
```
In the previous Amazon Pay version, amazon content was rendered within the container, so this was not an issue and all worked properly. Will this be fixed?

**Comment by ggarcia-bc** [comment]
Thanks for creating this issue @spire-mike, this is work in progress

**Comment by mauricio-sg** [comment]
The issue has been solved in `v1.300.1`
